### ISSUE TITLE
UnhandledPromiseRejectionWarning: #<Object> warning

### ISSUE THREAD

**Issue by sunilgautam** [zerodha/kiteconnectjs#35]
Getting following warning on calling generateSession function with invalid code.

```
(node:5690) UnhandledPromiseRejectionWarning: #<Object>
(Use `node --trace-warnings ...` to show where the warning was created)
(node:5690) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
(node:5690) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Here is [runkit link](https://runkit.com/sunilgautam/5f7ab7e7972ffa001ba1be5e) to reproduce this issue.

**Comment by sanchitrk** [comment]
The issue is with Inconsistent handling of Promise states, when a Promise is rejected & due to the async nature of JS the code keeps on exec, until the unhandled rejection bursts out the JS call stack.

I have created a fork that solves the above issue https://github.com/sanchitrk/kiteconnectjs. You can also install the package from github repository:-
create the `.npmrc` file in your node project and add the following line:

`@sanchitrk:registry=https://npm.pkg.github.com`

so that when you do `npm install @sanchitrk/kiteconnect@3.1.2` it installs from the above npm registry.

**Comment by arjun-k-r** [comment]
@sanchitrk 
i have created .npmrc file and tried to install the same, but ran into problem, below is error log, Kindly help me to resolve this.


D:\WORK\FIRETICK>npm install @socialstockdev/kiteconnect@3.1.3
npm ERR! code E401
npm ERR! Unable to authenticate, need: Basic realm="GitHub Package Registry"

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Arjun\AppData\Roaming\npm-cache\_logs\2021-02-11T04_42_57_553Z-debug.log

**Comment by sanchitrk** [comment]
Hi please follow the instructions [here](https://github.com/socialstockdev/kiteconnectjs/issues/2)

**Comment by sunilgautam** [comment]
@sanchitrk I tried the fork and It is solving the issue. Thanks

**Comment by sanchitrk** [comment]
@sunilgautam glad it helped.

**Comment by vividvilla** [comment]
Apologies for leaving this late, this has been fixed now on [v3.2.0](https://www.npmjs.com/package/kiteconnect/v/3.2.0). You can upgrade to latest version by running 

```
# Via NPM
npm update kiteconnect@3.2.0
# Via Yarn
yarn upgrade kiteconnect@3.2.0
```
